### PR TITLE
feat(website): 09/2025 devlog

### PR DESCRIPTION
**Pull request by jamilbk**
First in a series of new monthly devlog posts to summarize what we've shipped over the previous month.

Intentionally detailed and engineering focused - added a new `Engineering` section to the blog.

### CODE REVIEW

**Comment by jamilbk** [comment]
Tagging reviews on this PR, but I don't want to over-solicit reviews on marketing content in general, since part of it will be shipping quickly to see what works vs doesn't. (minor) Bugs here are low cost and easy to fix.

File: website/src/app/blog/devlog/2025-09/readme.mdx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+DNS responses are now cached for exactly as long as the authoritative nameserver specifies, reducing unnecessary queries while ensuring timely updates.
+
+## Performance Under Load
+
+We optimized the event-loop to prevent starvation of lower-priority inputs.[^10347]
+The tunnel now batch-processes input from all sources rather than prioritizing high-traffic sources to completion.
+This prevents timeout checks and DNS resolution from being delayed when UDP sockets or the TUN device are extremely busy.
+
+Linux systems got a significant performance boost through better socket buffer management.[^10349]
+Clients and Gateways now attempt to set receive and send buffer maximums to 4MB at startup.
```
Comment (thomaseizinger) [inline website/src/app/blog/devlog/2025-09/readme.mdx]: The 4MB reference is wrong. It is 128 for receive and 16 for send.

File: website/src/app/blog/devlog/2025-09/readme.mdx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+The default 200KB buffers were causing packet drops during high throughput scenarios, directly correlating with buffer errors visible in `nstat`.
+
+We also limited the number of optimistic ICE candidates to prevent CPU spikes.[^10367]
+When clients advertise many IPv6 addresses, the previous unlimited candidate generation could cause performance issues.
+Optimistic candidates are now disabled entirely for IPv6 and limited to 2 for IPv4, focusing on the scenarios where they're most beneficial.
+
+## Relay Infrastructure Advances
+
+Our relay infrastructure received important eBPF improvements.
+The relay can now handle relay-to-relay candidate pairs, which occur when both Client and Gateway allocate from the same relay.[^10286]
```
Comment (thomaseizinger) [inline website/src/app/blog/devlog/2025-09/readme.mdx]:
```suggestion
The relay can now handle relay-to-relay candidate pairs in the eBPF kernel, which occur when both Client and Gateway allocate from the same relay.[^10286]
```

File: website/src/app/blog/devlog/2025-09/readme.mdx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+We restructured the eBPF code into modular components that handle all cross-stack translation cases, including IPv4↔IPv6 transformations.
+A comprehensive integration test with double symmetric NAT validates this behavior in continuous integration.
+
+We also fixed several edge cases in the eBPF layer, including properly handling DNS replies[^10330] and re-populating the channel map when TURN channels are refreshed.[^10291]
+
+## Platform-Specific Improvements
+
+The Android client now launches authentication in a CustomTab instead of the default browser.[^10371]
+This fixes a Firefox bug where only the first browser tab could intercept the custom URI scheme, making subsequent sign-ins fail until users manually closed the old tab.
+CustomTabs ensure only one sandboxed authentication instance exists at a time.
```
Comment (thomaseizinger) [inline website/src/app/blog/devlog/2025-09/readme.mdx:73]:
```suggestion
CustomTabs ensure only one sandboxed authentication instance exists at a time.
It also results in a nice UX improvement where the "You have been signed-in" tabs are no longer lingering around.
```

I think that is quite nice. It would also be awesome if we could achieve the same thing on desktop somehow.

**Review by thomaseizinger — APPROVED** [review]
Nice work! How long did it take you to edit it? Just curious.

File: website/src/app/blog/devlog/2025-09/readme.mdx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+DNS responses are now cached for exactly as long as the authoritative nameserver specifies, reducing unnecessary queries while ensuring timely updates.
+
+## Performance Under Load
+
+We optimized the event-loop to prevent starvation of lower-priority inputs.[^10347]
+The tunnel now batch-processes input from all sources rather than prioritizing high-traffic sources to completion.
+This prevents timeout checks and DNS resolution from being delayed when UDP sockets or the TUN device are extremely busy.
+
+Linux systems got a significant performance boost through better socket buffer management.[^10349]
+Clients and Gateways now attempt to set receive and send buffer maximums to 4MB at startup.
```
Comment (jamilbk) [inline website/src/app/blog/devlog/2025-09/readme.mdx]:
```suggestion
Clients and Gateways now attempt to set UDP receive and send buffer maximums to 128 MB and 16 MB respectively at startup.
```

**Comment by jamilbk** [comment]
> Nice work! How long did it take you to edit it? Just curious.

Probably about 1/2 hour on the generation of content, and another 1/2 hour integrating it and adjusting visual / layout things.
